cmd/kube-apiserver/app/server.go

```diff
@@ -455,6 +456,7 @@ func (s *APIServer) Run(_ []string) error {
 						glog.Infof("Using self-signed cert (%s, %s)", s.TLSCertFile, s.TLSPrivateKeyFile)
 					}
 				}
+				_ = systemd.SdNotify("READY=1\n")
```

Comment:
Looks like it might be easier to check for the existence of the `NOTIFY_SOCKET` environment variable to tell if you are running under systemd.